lib/player.js

```diff
@@ -3242,6 +3242,83 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
     }
   }
 
+  /**
+   * Return a list of image tracks that can be switched to.
+   *
+   * <p>
+   * If the player has not loaded content, this will return an empty list.
+   *
+   * @return {!Array.<shaka.extern.Track>}
+   * @export
+   */
+  getImageTracks() {
+    if (this.manifest_) {
+      const imageStreams = this.manifest_.imageStreams;
+      const StreamUtils = shaka.util.StreamUtils;
+      return imageStreams.map((image) => StreamUtils.imageStreamToTrack(image));
+    } else {
+      return [];
+    }
+  }
+
+  /**
+   * Return a Thumbnail object from a image track Id and time.
+   *
+   * <p>
+   * If the player has not loaded content, this will return a null.
+   *
+   * @param {number} trackId
+   * @param {number} time
+   * @return {!Promise.<?shaka.extern.Thumbnail>}
+   * @export
+   */
+  async getThumbnails(trackId, time) {
+    if (this.manifest_) {
+      const imageStream = this.manifest_.imageStreams.find(
+          (stream) => stream.id == trackId);
+      if (!imageStream) {
+        return null;
+      }
+      if (!imageStream.segmentIndex) {
+        await imageStream.createSegmentIndex();
+      }
+      const referencePosition = imageStream.segmentIndex.find(time);
+      if (referencePosition == null) {
+        return null;
+      }
+      const reference = imageStream.segmentIndex.get(referencePosition);
+      const match = /(\d+)x(\d+)/.exec(imageStream.tilesLayout);
```

Comment:
nit: Please add a comment for the regex. Not everyone reads them at a glance, but everyone deserves to easily see what they're doing :)

Reply:
Done!